vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go

```diff
@@ -334,8 +338,8 @@ func handleEndpointExtensions(pluginHandler PluginHandler, cmdArgs []string) err
 
 	// attempt to find binary, starting at longest possible name with given cmdArgs
 	for len(remainingArgs) > 0 {
-		path, err := pluginHandler.Lookup(fmt.Sprintf("kubectl-%s", strings.Join(remainingArgs, "-")))
```

Comment:
With `prefixes` being part of `defaultPluginHandler` struct you won't have to change this bit.